### DISCUSSION TITLE
Unable to access collection hook fields through the graphql API. Any workarounds?

### DISCUSSION THREAD

**Discussion by richardvanbergen** [payloadcms/payload#392] [Feature Requests & Ideas]
Currently, `payload-plugin-cloud-storage` uses a collection `afterRead` hook to add the cloud URL to both the main document and the `sizes`. I wasn't using the graphQL in production so I didn't notice but this method means that it's not available through the GraphQL api.

I could add a virtual field and a field hook but then I can't adjust the `sizes` array. I can think of a couple of horrible workarounds like storing them all as JSON but I can't think of a clean implementation. Any suggestions?

**Comment by jmikrut** [comment]
Hey @richardvanbergen — I see the issue here.

Question for you. I recall that we added the `url` field to base `upload` docs as well as each of the sizes in the `sizes` object. Is it possible that you could format those URL fields via a document `afterRead` hook?

Alternatively, we could come up with some type of API that lets you merge in or provide your own fields to the `sizes` base fields. This shouldn't be too hard to do.

What do you think?

**Reply by richardvanbergen** [reply]
> Is it possible that you could format those URL fields via a document afterRead hook?

Unless I'm misunderstanding, I think that is what I currently do? The issue is that I don't think that information is exposed via the GraphQL API?

> Alternatively, we could come up with some type of API that lets you merge in or provide your own fields to the sizes base fields. This shouldn't be too hard to do.

Yes I think that would be ideal. I'm not sure what a nice shape for that would be.

**Reply by jmikrut** [reply]
Ah, ok. Those fields are indeed exposed via GraphQL.

Running the following query against a `media` collection returns URL fields as expected:

```graphql
  query {
    allMedia {
      docs {
        filename
        url
        sizes {
          tablet {
            url
          }
        }
      }
    }
  }
```

Is this what you're referring to as well?

**Reply by richardvanbergen** [reply]
OK I'll have to check that out as it's not the behaviour I was experiencing. 🤔
